models/models/markov_chain/markov_chain.py

```diff
@@ -92,11 +92,13 @@ def prepare_data(self, midi_file: Path) -> tuple[Any, Any]:
         return 0, 0
 
     def save(self, path: Path) -> None:
-        np.save(path, np.asarray(self.probabilities))
+        np.savez(path, probabilities=np.asarray(self.probabilities, dtype=object),
+                 tokens=np.asarray(self.tokens_list, dtype=object))
 
     def load(self, path: Path) -> None:
-        path = path if path.name.endswith('.npy') else path.with_suffix('.npy')
```

Comment:
Please bring back that line (but with `npz`) it is needed to let backend correctly load